momepy/preprocessing.py

```diff
@@ -740,3 +746,327 @@ def _get_extrapolated_line(coords, tolerance, point=False):
+            edges.geometry, predicate="covered_by"
+        )
+    else:
+        touching = gpd.sjoin(edges, rab_multipolygons, op="touches")
+        edges_idx, rabs_idx = rab_multipolygons.sindex.query_bulk(
+            edges.geometry, op="covered_by"
+        )
+    idx_drop = edges.index.take(edges_idx)
+    touching_idx = touching.index
+    ls = list(set(touching_idx) - set(idx_drop))
+
+    incoming = touching.loc[ls]
+
+    # figuring out which ends of incoming edges needs to be connected to the center_pt
+    incoming["first_pt"] = incoming.geometry.apply(lambda x: Point(x.coords[0]))
+    incoming["dist_fisrt_pt"] = incoming.center_pt.distance(incoming.first_pt)
+    incoming["last_pt"] = incoming.geometry.apply(lambda x: Point(x.coords[-1]))
+    incoming["dist_last_pt"] = incoming.center_pt.distance(incoming.last_pt)
+    lines = []
+    for i, row in incoming.iterrows():
+        if row.dist_fisrt_pt < row.dist_last_pt:
+            lines.append(LineString([row.first_pt, row.center_pt]))
+        else:
+            lines.append(LineString([row.last_pt, row.center_pt]))
+    incoming["line"] = gpd.GeoSeries(lines, index=incoming.index, crs=edges.crs)
+
+    # checking in there are more than one incoming lines arriving to the same point
+    # which would create several new lines
+    incoming["line_wkt"] = incoming.line.to_wkt()
+    grouped_lines = incoming.groupby(["line_wkt"])["line_wkt"]
+    count_s = grouped_lines.count()
+
+    # separating the incoming roads that come on their own to those that come in groups
+    filter_count_one = pd.DataFrame(count_s[count_s == 1])
+    filter_count_many = pd.DataFrame(count_s[count_s > 1])
+    incoming_ones = pd.merge(
+        incoming, filter_count_one, left_on="line_wkt", right_index=True, how="inner"
+    )
+    incoming_many = pd.merge(
+        incoming, filter_count_many, left_on="line_wkt", right_index=True, how="inner"
+    )
+    incoming_many_reduced = _coins_filtering_many_incoming(
+        incoming_many, angle_threshold=angle_threshold
+    )
+
+    incoming_all = gpd.GeoDataFrame(
+        pd.concat([incoming_ones, incoming_many_reduced]), crs=edges.crs
+    )
+
+    return incoming_all, idx_drop
+
+
+def _ext_lines_to_center(edges, incoming_all, idx_out):
+    """
+    Extends the Linestrings geometries to the centerpoint defined by
+    _rabs_center_points. Also deletes the lines that originally defined the roundabout.
+    Creates a new column labled with the 'rab' number.
+
+    Returns
+    -------
+    GeoDataFrame
+        GeoDataFrame of with updated geometry
+    """
+
+    incoming_all["geometry"] = incoming_all.apply(
+        lambda row: linemerge([row.geometry, row.line]), axis=1
+    )
+
+    # deleting the original round about edges
+    new_edges = edges.drop(idx_out, axis=0)
+
+    # creating a unique label for returned gdf
+    ls = ["rab_" + x for x in incoming_all.index_right.apply(str)]
+    incoming_label = pd.Series(ls, index=incoming_all.index)
```

Comment:
```suggestion
    incoming_label = "rab_" + incoming_all.index_right.astype(str)
```

this is optimised in pandas, list comprehension will be slower

Reply:
I knew there had to be something better! thanks